pkg/azurefile/utils.go

```diff
@@ -43,15 +43,17 @@ func newLockMap() *lockMap {
 }
 
 // LockEntry acquires a lock associated with the specific entry
-func (lm *lockMap) LockEntry(entry string) {
+func (lm *lockMap) LockEntry(entry string) bool {
 	lm.Lock()
-	// check if entry does not exists, then add entry
-	if _, exists := lm.mutexMap[entry]; !exists {
-		lm.addEntry(entry)
-	}
+	defer lm.Unlock()
 
-	lm.Unlock()
+	// check if entry exists, then it is not possible to acquire lock
+	if _, exists := lm.mutexMap[entry]; exists {
+		return false
+	}
+	lm.addEntry(entry)
 	lm.lockEntry(entry)
```

Comment:
@andyzhangx Should we use `sets.String` instead of `mutexMap map[string]*sync.Mutex`? It will simplify the logic. 

Similar to: https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/2014365ea7b3405e024be1d9552aa30d44da4a8b/pkg/common/volume_lock.go#L32

Reply:
Please ignore. I'm still updating this.